metrics/prometheus_reporter.go

```diff
@@ -344,6 +344,11 @@ func (p *PrometheusReporter) ReportGauge(metric string, labels map[string]string
 	return constants.ErrMetricNotKnown
 }
 
+// ReportEvent is not implemented by PrometheusReporter
+func (p *PrometheusReporter) ReportEvent(title string, text string) error {
+	return nil
```

Comment:
Since Prometheus doesn't provide a way to report events, we should at least log when someone is trying to use this method (throw a warning or even an error log).

Reply:
Actually, returning an error explaining the feature is not supported might be even better 🤔

Reply:
Agreed! Pushed the change to return an error.